### PR TITLE
chore: Rounding, frames and radio buttons of the booking forms not consistent

### LINKED ISSUE #7508: Rounding, frames and radio buttons of the booking forms not consistent
1. The rounding of the frame for the phone number is different from the other fields.
2. For the "Name" field, the frame is much thinner than for the other input fields and is displayed only in the branding color. For the other fields there is an additional black frame next to the branding color.
3. The radio buttons do not adopt the branding color, they just stay black.

Uploading Bildschirmaufnahme 2023-03-04 um 02.14.26.mov…

### PR DESCRIPTION
**Pull request by gitstart-calcom**
Fixes #7508 

## Demo
https://www.loom.com/share/f020b84243ca48928f6d44d8b9df6f92

Steps to reproduce: 
1. Edit some event-type 
2. Click on advanced settings and add all the options
3. Go to settings/appearance and change the brand color
4. Try to schedule some meeting

FIgma with the updated design: https://www.figma.com/file/9MOufQNLtdkpnDucmNX10R/%E2%9D%96-Cal-DS?node-id=309-44674&t=XVOdTk8kjKnXp6ti-0

### CODE REVIEW

File: packages/app-store/routing-forms/components/react-awesome-query-builder/widgets.tsx [hunk 1]
```diff
@@ -129,7 +129,7 @@ function NumberWidget({ value, setValue, ...remainingProps }: TextLikeComponentP
       type="number"
       labelSrOnly={remainingProps.noLabel}
       containerClassName="w-full"
-      className="focus:border-brand-default bg-default dark:bg-muted  border-default disabled:bg-emphasis focus:ring-brand block w-full rounded-md text-sm disabled:hover:cursor-not-allowed"
+      className="bg-default dark:bg-muted border-default disabled:bg-emphasis block w-full rounded-md text-sm focus-within:border-neutral-300 focus-within:outline-none focus-within:ring-2 focus-within:ring-neutral-800 disabled:hover:cursor-not-allowed"
```
Comment (PeerRich) [inline packages/app-store/routing-forms/components/react-awesome-query-builder/widgets.tsx:132]: @sean-brydon i think this should be using our color tokens no?

File: package.json [hunk 2]
```diff
@@ -96,6 +96,7 @@
     "city-timezones": "^1.2.1",
     "eslint": "^8.34.0",
     "lucide-react": "^0.130.0",
+    "react-phone-input-2": "^2.15.1",
```
Comment (PeerRich) [inline package.json:99]: @zomars is this right here in root? should this not be in packages/ or somewhere else?

File: packages/features/form-builder/Components.tsx [hunk 3]
```diff
@@ -314,7 +317,8 @@ export const Components: Record<BookingFieldType, Component> = {
                         type="radio"
                         disabled={readOnly}
                         name={name}
-                        className="dark:bg-darkgray-300 border-subtle text-emphasis h-4 w-4 focus:ring-black ltr:mr-2 rtl:ml-2"
+                        className="dark:bg-darkgray-300 border-default focus:border-default focus:ring-emphasis h-4 w-4 cursor-pointer border focus:ring-2 ltr:mr-2 rtl:ml-2"
```
Comment (PeerRich) [inline packages/features/form-builder/Components.tsx:320]: same here, shouldnt this be tokens?

File: packages/features/form-builder/Components.tsx [hunk 4]
```diff
@@ -314,7 +317,8 @@ export const Components: Record<BookingFieldType, Component> = {
...
+                        style={{ color: `${user?.brandColor}` }}
```
Comment (PeerRich) [inline packages/features/form-builder/Components.tsx:321]: this should be a classname too, no?

File: packages/ui/components/form/checkbox/Checkbox.tsx [hunk 5]
```diff
@@ -53,13 +55,11 @@ const CheckboxField = forwardRef<HTMLInputElement, Props>(
...
                     className={classNames(
-                      "text-primary-600 focus:ring-primary-500 border-default bg-default h-4 w-4 rounded ltr:mr-2 rtl:ml-2",
-                      !error && disabled
-                        ? "bg-gray-300 checked:bg-gray-300"
-                        : "hover:bg-subtle checked:bg-gray-800",
+                      "border-default h-4 w-4 rounded ltr:mr-2 rtl:ml-2",
                       error && "border-red-800 checked:bg-red-800 hover:bg-red-400",
                       rest.className
                     )}
+                    style={{ color: `${user?.brandColor}` }}
```
Comment (PeerRich) [inline packages/ui/components/form/checkbox/Checkbox.tsx:62]: are we sure we want brandColor here?

File: packages/app-store/routing-forms/components/react-awesome-query-builder/widgets.tsx (hunk 1 above)
Comment (sean-brydon) [reply packages/app-store/routing-forms/components/react-awesome-query-builder/widgets.tsx:132]: Yes @gitstart-calcom  please convert this to our [tokens](https://www.figma.com/file/9MOufQNLtdkpnDucmNX10R/%E2%9D%96-Cal-DS?type=design&t=RpKFVybDZL62nMnk-6)

File: packages/features/form-builder/Components.tsx [hunk 6]
```diff
@@ -240,6 +240,7 @@ export const Components: Record<BookingFieldType, Component> = {
     propsType: "multiselect",
     factory: ({ options, readOnly, setValue, value }) => {
       value = value || [];
+      const { data: user } = trpc.viewer.me.useQuery();
```
Comment (sean-brydon) [inline packages/features/form-builder/Components.tsx:243]:
```suggestion
```

File: packages/features/form-builder/Components.tsx [hunk 7]
```diff
@@ -255,7 +256,8 @@ export const Components: Record<BookingFieldType, Component> = {
                     }
                     setValue(newValue);
                   }}
-                  className="dark:bg-darkgray-300 border-subtle text-emphasis h-4 w-4 rounded focus:ring-black ltr:mr-2 rtl:ml-2"
+                  className="border-subtle h-4 w-4 rounded ltr:mr-2 rtl:ml-2"
+                  style={{ color: `${user?.brandColor}` }}
```
Comment (sean-brydon) [inline packages/features/form-builder/Components.tsx:260]:
```suggestion
```

File: packages/features/form-builder/Components.tsx [hunk 8]
```diff
@@ -255,7 +256,8 @@ export const Components: Record<BookingFieldType, Component> = {
                     }
                     setValue(newValue);
                   }}
-                  className="dark:bg-darkgray-300 border-subtle text-emphasis h-4 w-4 rounded focus:ring-black ltr:mr-2 rtl:ml-2"
+                  className="border-subtle h-4 w-4 rounded ltr:mr-2 rtl:ml-2"
```
Comment (sean-brydon) [inline packages/features/form-builder/Components.tsx:259]:
```suggestion
                  className="border-subtle h-4 w-4 rounded ltr:mr-2 rtl:ml-2 text-brand"
```

File: packages/features/form-builder/Components.tsx [hunk 9]
```diff
@@ -301,6 +303,7 @@ export const Components: Record<BookingFieldType, Component> = {
           });
         }
       }, [options, setValue, value]);
+      const { data: user } = trpc.viewer.me.useQuery();
```
Comment (sean-brydon) [inline packages/features/form-builder/Components.tsx:306]:
```suggestion

```

File: packages/features/form-builder/Components.tsx (hunk 3 above)
Comment (sean-brydon) [inline packages/features/form-builder/Components.tsx:320]:
```suggestion
                        className="bg-emphasis text-brand border-default focus:border-default focus:ring-emphasis h-4 w-4 cursor-pointer border focus:ring-2 ltr:mr-2 rtl:ml-2"
```

File: packages/features/form-builder/Components.tsx (hunk 4 above)
Comment (sean-brydon) [inline packages/features/form-builder/Components.tsx:321]:
```suggestion
```

File: packages/ui/components/form/checkbox/Checkbox.tsx [hunk 10]
```diff
@@ -16,6 +17,7 @@ type Props = InputHTMLAttributes<HTMLInputElement> & {
 const CheckboxField = forwardRef<HTMLInputElement, Props>(
   ({ label, description, error, disabled, ...rest }, ref) => {
     const descriptionAsLabel = !label || rest.descriptionAsLabel;
+    const { data: user } = trpc.viewer.me.useQuery();
```
Comment (sean-brydon) [inline packages/ui/components/form/checkbox/Checkbox.tsx:20]:
```suggestion

```

**Review by sean-brydon — DISMISSED** [review]
Thanks @gitstart-calcom we need to use our tokens in this so our colours are consitant throughout the app. 

I've not visually reviewed this so i dont know the classes are correct. Please use our `brand` class names to style with the users brand colors. 

If youre more intrested in how this works look at our figma or take a look at `cal.com/packages/config/tailwind-preset.js` to see all of our available tokens <3

File: packages/ui/components/form/checkbox/Checkbox.tsx (hunk 5 above)
Comment (gitstart-calcom) [reply packages/ui/components/form/checkbox/Checkbox.tsx:62]: Or is the Checkbox not supposed to be in brandColor?

Comment (sean-brydon) [reply packages/ui/components/form/checkbox/Checkbox.tsx:62]: > Do you mean we should move this brandColor to className?

we should move it to the classNames i have mentioned below :)

**Comment by gitstart-calcom** [comment]
Hi @sean-brydon @PeerRich, PR has been updated, please take a look
https://www.loom.com/share/f020b84243ca48928f6d44d8b9df6f92

File: packages/features/form-builder/Components.tsx [hunk 11]
```diff
@@ -134,7 +134,7 @@ export const Components: Record<BookingFieldType, Component> = {
       const { t } = useLocale();
       value = value || [];
       const inputClassName =
-        "dark:placeholder:text-darkgray-600 focus:border-brand-default border-subtle  block w-full rounded-md border-default text-sm focus:ring-black disabled:bg-emphasis disabled:hover:cursor-not-allowed dark:bg-transparent dark:selection:bg-green-500 disabled:dark:text-subtle";
+        "dark:placeholder:text-darkgray-600 focus:border-emphasis border-subtle block w-full rounded-md border-default text-sm focus:ring-black disabled:bg-emphasis disabled:hover:cursor-not-allowed dark:selection:bg-green-500 disabled:dark:text-subtle bg-default";
```
Comment (sean-brydon) [inline packages/features/form-builder/Components.tsx:137]: NIT - we have tokens for text-darkgray-600 now 

placeholder:text-muted should work for dark/lightmode here

File: packages/features/form-builder/Components.tsx [hunk 12]
```diff
@@ -310,7 +310,7 @@ export const Components: Record<BookingFieldType, Component> = {
                         type="radio"
                         disabled={readOnly}
                         name={name}
-                        className="dark:bg-darkgray-300 border-subtle text-emphasis h-4 w-4 focus:ring-black ltr:mr-2 rtl:ml-2"
+                        className="dark:checked:bg-brand-default dark:bg-darkgray-100 dark:hover:bg-subtle dark:checked:hover:bg-brand-default focus:border-brand-default focus:ring-brand-default border-emphasis h-4 w-4 cursor-pointer text-[--cal-brand] focus:ring-2 ltr:mr-2 rtl:ml-2"
```
Comment (sean-brydon) [inline packages/features/form-builder/Components.tsx:313]:
```suggestion
                        className="dark:checked:bg-brand-default dark:bg-darkgray-100 dark:hover:bg-subtle dark:checked:hover:bg-brand-default focus:border-brand-default focus:ring-brand-default border-emphasis h-4 w-4 cursor-pointer text-brand focus:ring-2 ltr:mr-2 rtl:ml-2"
```

NIT I think we need to double check the classes here - Not using tokens in a few places

**Review by sean-brydon — DISMISSED** [review]
LGTM NITs dont need to be adressed but it would be ideal :) 

Just waiting for yarn.lock to be removed from this PR

**Comment by PeerRich** [comment]
> LGTM NITs dont need to be adressed but it would be ideal :)
> 
> Just waiting for yarn.lock to be removed from this PR

are you sure tho? We're adding a new package

File: packages/features/form-builder/Components.tsx (hunk 12 above)
Comment (gitstart-calcom) [reply packages/features/form-builder/Components.tsx:313]: When using `text-brand` the checkbox turns white. Please check this out
https://www.loom.com/share/2f0bdc127bdb4221b96660f459d6d46c

Comment (sean-brydon) [reply packages/features/form-builder/Components.tsx:313]: Interesting okay - lets just leave it as it is for now and stick with the css variable. 

I will look into this in another PR

**Comment by sean-brydon** [comment]
ALso apologies i didnt realise there was a new package being added here! Looks like we may need the lock file after all! Sorry!

**Comment by gitstart-calcom** [comment]
Added the lock file back and also made the requested changes @sean-brydon

**Review by zomars — DISMISSED** [review]
Seems like the phone input is broken in the new booker. cc @JeroenReumkens can you help debug this? 

image

**Comment by JeroenReumkens** [comment]
@gitstart  @zomars  This phone input issue seems to be related to an overflow issue that I also had when making the booker work as an embed. I fixed it in that (draft) PR: #9023 

For now it's safe to ignore the overflow issue. I will test it once that PR is merged and make sure it works, and fix it otherwise ❤️

**Comment by gitstart-calcom** [comment]
> @gitstart @zomars This phone input issue seems to be related to an overflow issue that I also had when making the booker work as an embed. I fixed it in that (draft) PR: #9023
> 
> For now it's safe to ignore the overflow issue. I will test it once that PR is merged and make sure it works, and fix it otherwise ❤️

Hi @zomars, what do you think?

**Review by JeroenReumkens — APPROVED** [review]
Great changes @gitstart-calcom! I approved them 🙏

**Review by PeerRich — APPROVED** [review]

**Review by zomars — APPROVED** [review]
Ship it

**Review by zomars — APPROVED** [review]

**Review by sean-brydon — APPROVED** [review]

**Pull request merged**